### DISCUSSION TITLE
RealmResourceProvider spi-id

### DISCUSSION THREAD

**Discussion by illuka92** [keycloak/keycloak#23693] [Q&A]
Good morning everyone, I have the following problem: 
I created an extension for my keycloak 20.0.5 that implements RealmResourceProviderFactory, now, what value does the spi-id have to use to be able to insert my custom property into keycloak.conf?

For example I want to create the URL variable in my component (id: proxy-test) , I know I will have to insert:
spi-<spi-id>-proxy-test-URL=<value>

which is the value of spi-id? I tried with realm-resouce ; resource ; realms-resource; resources and many other combinations but nothing

**Comment by kwame-mintah** [comment]
@illuka92, I don't know what it should be, but check the providers list within the Keycloak Admin UI. Navigate to the 'Provider info' tab, next to 'Realm info'. I cannot see any of the SPI ID's you have listed in your post.

e.g. To override the emailSender SPI with your own provider
```bash
--spi-email-sender-provider=<PROVIDER-NAME>
```

If you have loaded the custom provider successfully into Keycloak, will be shown next to the SPI.

**Comment by danielFesenmeyer** [comment]
You can find it here:  https://github.com/keycloak/keycloak/blob/21a23ace1d0aaada4825f4627b9bc3835d190ff3/server-spi-private/src/main/java/org/keycloak/services/resource/RealmResourceSPI.java#L42

It's `realm-restapi-extension`.

**Comment by illuka92** [comment]
Hi everyone! T
hank you so much for your support! 

I actually solved it this way: I loaded the component onto keycloak and noticed that if I go to realm master --> providers, I find my spi and next to it the name used by keycloak for that type.

**Reply by kwame-mintah** [reply]
Glad to hear you found the SPI.
